Styhead/rz cmn: Support common binary bl2/bl31

ThanhNguyenRVC opened the pull request:
Changes include:
- Support common BL2/BL31 binary for RZG2L-SBC, RZG2L-EVK, RZV2L-EVK, RZV2H-EVK.
- Support ATF device tree build option for supported devices (make PLAT=cmn BOARD=rz_cmn dtbs)

vudangRVC commented on plat/renesas/rz/common/bl2_plat_mem_params_desc.c:
```diff
@@ -8,14 +8,10 @@
-#include <rzg2l_def.h>
-#endif
-
+#include <rzcmn_def.h>
```
"rzcmn" refers to all RZ SoCs we support. If you split the definition for each SoC, please assign the appropriate name to each one. 

In this case, I believe "rzg2l_def" is the correct name, as it covers both RZG2L and RZV2L SoCs.

If some are common, put them in "rzcmn_def"

vudangRVC replied:
Apply this change to all relevant files.

tiennguyenuh replied:
Alternatively, rzg2l_v2l_def.h is also an acceptable name for me.

ThanhNguyenRVC replied:
It is updated

vudangRVC commented on plat/renesas/rz/common/bl2_plat_setup.c:
```diff
@@ -89,20 +119,28 @@ void bl2_el3_early_platform_setup(u_register_t arg1, u_register_t arg2,
+	bootrom_param0 = arg1;
+
+	if ((bootrom_param0 == RZG2L_BL2_BASE)) {
+		dtb_base = RZCMN_DTB_BASE;
```
You should change this macro to RZG2L_DTB_BASE

ThanhNguyenRVC replied:
It is renamed

vudangRVC commented on plat/renesas/rz/common/bl2_plat_setup.c:
```diff
@@ -18,55 +18,85 @@
-};
-
-static console_t rzg2l_bl31_console;
+/* Global variable where runtime BL2 base is stored */
```
Please add more comments to describe all params from BL1.
It explains why we use the first param

ThanhNguyenRVC replied:
I have updated the description. Attached below is information about all the parameters across the SoCs.

<img width="990" height="494" alt="image" src="https://github.com/user-attachments/assets/172c4d2f-9841-45ee-9388-cb6a74b0dcaf" />

<img width="800" height="378" alt="image" src="https://github.com/user-attachments/assets/c695ed92-f74a-47c4-acef-2e7849b12505" />

vudangRVC replied:
It's already defined here: https://github.com/Renesas-SST/rz-atf/pull/9/files#diff-9676a640a328ab5a66125821a26ca84eabf8b0c27179974ca147acfcca39d051R36

I'd like to close this thread

vudangRVC commented on plat/renesas/rz/common/bl31_plat_setup.c:
```diff
@@ -14,52 +14,167 @@
+		.board_info_qspi_offset = BOARD_INFO_QSPI_OFFSET,
+
+		.enable_tzc_setup  = true,
+		.enable_pwrc_setup = false,
```
RZCMN -> RZG2L

tiennguyenuh replied:
Alternatively, RZG2L_V2L_ is also an acceptable name for me.

ThanhNguyenRVC replied:
It is renamed

vudangRVC commented on plat/renesas/rz/common/bl31_plat_setup.c:
```diff
@@ -14,52 +14,167 @@
 
 void bl31_plat_arch_setup(void)
 {
+
```
This line makes no sense, maybe mistake?

ThanhNguyenRVC replied:
It is removed

tiennguyenuh commented on plat/renesas/rz/board/rz_cmn/rz_board.mk:
```diff
@@ -0,0 +1,13 @@
+#
+# Copyright (c) 2023, Renesas Electronics Corporation. All rights reserved.
```
Change 2023 to 2025

ThanhNguyenRVC replied:
It is updated.

vudangRVC commented on plat/renesas/rz/common/board_info.c:
```diff
@@ -98,3 +98,27 @@ void get_board_info_string(uintptr_t flash_base, size_t board_info_offset, size_
+ * for selecting the correct OTP base depending on the SoC.
+ *
+ */
+void get_chipid(uintptr_t otp_base, uint32_t *chipid)
```
This comment section is duplicated to plat/renesas/rz/common/include/board_info.h.

Additionally, the comment mentions "read_chipid" but declaration is "get_chipid", mistake?

ThanhNguyenRVC replied:
I have update the description

vudangRVC commented on plat/renesas/rz/common/drivers/cpg.c:
```diff
@@ -5,511 +5,77 @@
-
 #define CPG_SEL_PLL1_ON_OFF					(0)
 #define CPG_SEL_PLL2_1_ON_OFF				(1)
+
```
This line makes no sense

ThanhNguyenRVC replied:
It is removed

vudangRVC commented on plat/renesas/rz/common/drivers/cpg.c:
```diff
@@ -731,135 +334,182 @@ static void cpg_selector_on_off(uint32_t sel, uint8_t flag)
 }
 
 /* It is assumed that the PLL has stopped by the time this function is executed. */
 static void cpg_pll_setup(void)
```
Check the flow in this functions. Seems like you changed it

ThanhNguyenRVC replied:
I explain the old flow of the PLL setup functions in the image below:

<img width="1280" height="720" alt="cpg_old_flow" src="https://github.com/user-attachments/assets/987ab918-fa88-47ae-bc98-d9c18786ec46" />

In the new flow, only steps *2 and *3 are required (start the PLL clock and ensure it has stabilized).
The step *1 in the old flow is redundant and should be removed.

<img width="911" height="713" alt="image" src="https://github.com/user-attachments/assets/d45514f3-8606-4c6d-a2dc-5879b0e61290" />

tiennguyenuh commented on plat/renesas/rz/common/include/platform_def.h:
```diff
@@ -53,20 +55,41 @@
-#define BL2_LIMIT				(0x0002F000)
+#define RZV2L_BL2_BASE				(0x00013000)
+#define RZV2L_BL2_LIMIT				(0x0002F000)
 #endif
```
These definitions are the same, we can use common RZG2L_V2L_ name

ThanhNguyenRVC replied:
As RZG2L and V2L share the common value, I removed for V2L

tiennguyenuh commented on plat/renesas/rz/common/include/rzcmn_def.h:
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020-2022, Renesas Electronics Corporation. All rights reserved.
```
Change to 2025

ThanhNguyenRVC replied:
It is updated

tiennguyenuh commented on plat/renesas/rz/common/include/xspi_regs_offset.h:
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022, Renesas Electronics Corporation. All rights reserved.
```
Change the year to 2025

ThanhNguyenRVC replied:
It is updated

tiennguyenuh commented on plat/renesas/rz/common/plat_security.c:
```diff
@@ -55,18 +70,117 @@ static const struct {
+	if ((board_id == BOARD_ID_RZG2L_EVK) ||
+		(board_id == BOARD_ID_RZG2L_SBC) ||
+		(board_id == BOARD_ID_RZV2L_EVK)) {
+		memcpy(sys_acctl, sys_cmn_acctl, sizeof(sys_cmn_acctl));
```
We need to modify the code if we add a new board here, right? Are there any other solutions for this?

ThanhNguyenRVC replied:
It is updated

tiennguyenuh commented on plat/renesas/rz/common/plat_storage.c:
```diff
@@ -374,10 +296,21 @@ void rz_io_setup(void)
 		register_io_dev_memmap(&memmap);
 		io_dev_open(memmap, 0, &memdrv_dev_handle);
 
+		if (board_id == BOARD_ID_RZV2H_EVK) {
```
Same here. We need to modify the code if we add a new board. Should we use a macro for the SoC instead? Please check with other files.

ThanhNguyenRVC replied:
It is updated

vudangRVC commented on plat/renesas/rz/soc/cmn/drivers/ddr/param_mc_cmn.c:
This file is empty

ThanhNguyenRVC replied:
I removed

ThanhNguyenRVC commented:
I have verified at lasted commit update:

[rzv2l-evk_cmn_bl2_bl31_boot_log_12092025.txt](https://github.com/user-attachments/files/22296187/rzv2l-evk_cmn_bl2_bl31_boot_log_12092025.txt)
[rzg2l-evk_cmn_bl2_bl31_boot_log_12092025.txt](https://github.com/user-attachments/files/22296188/rzg2l-evk_cmn_bl2_bl31_boot_log_12092025.txt)
[rzg2l-sbc_cmn_bl2_bl31_boot_log_12092025.txt](https://github.com/user-attachments/files/22296189/rzg2l-sbc_cmn_bl2_bl31_boot_log_12092025.txt)
[rzv2h-evk_cmn_bl2_bl31_boot_log_12092025.txt](https://github.com/user-attachments/files/22296190/rzv2h-evk_cmn_bl2_bl31_boot_log_12092025.txt)

tiennguyenuh commented on plat/renesas/rz/common/include/board_info.h:
```diff
@@ -16,6 +17,14 @@
+#define BOARD_ID_RZG2L_SBC				0x11
+#define BOARD_ID_RZV2L_EVK				0x20
+#define BOARD_ID_RZV2H_EVK				0x30
+
```
I do not recommend using a board-specific macro in BL2. It would be better to define it at the SoC level, as boards based on the same SoC typically share the same settings. Otherwise, adding a new board would require modifying and rebuilding BL2.

ThanhNguyenRVC replied:
I have supported SoC-based implementations with verification.

ThanhNguyenRVC committed:
renesas: rz-cmn: Support common binary bl2/bl31

Change includes:

- Support rzcmn TF-A platform
- Support common BL2/BL31 binary for RZG2L-SBC, RZG2L-EVK, RZV2L-EVK,
  RZV2H-EVK.
- Support ATF device tree build option for supported devices (make PLAT=cmn BOARD=rz_cmn dtbs)

Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>
